### PR TITLE
DateTime picker implementation

### PR DESCRIPTION
**Pull request by thecalcc**
STT-109

### CODE REVIEW

**Comment by thecalcc** [comment]
Note: z-index fixes aren't applied here so lint will fail

File: scripts/apps/authoring-react/fields/time copy/editor.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+type IProps = IEditorComponentProps<ITimeValueOperational, ITimeFieldConfig, ITimeUserPreferences>;
+
+export class Editor extends React.PureComponent<IProps> {
+    render() {
+        const Container = this.props.container;
+
+        return (
+            <Container>
+                <DateTimePicker
+                    dateFormat='MM/DD/YYYY'
```
Comment (tomaskikutis) [inline scripts/apps/authoring-react/fields/time copy/editor.tsx]: do not hardcode format - use the one from appConfig

**Review by tomaskikutis — COMMENTED** [review]
Do not use interfaces owned by time field(like ITimeFieldConfig). Add separate interface for a new field type. Even if it will be identical at the moment.

Rename "time copy" to something better

File: scripts/apps/authoring-react/fields/datetime/editor.tsx [hunk 2]
```diff
@@ -17,7 +18,7 @@ export class Editor extends React.PureComponent<IProps> {
         return (
             <Container>
                 <DateTimePicker
-                    dateFormat='MM/DD/YYYY'
+                    dateFormat={appConfig.shortDateFormat}
```
Comment (tomaskikutis) [inline scripts/apps/authoring-react/fields/datetime/editor.tsx]: use `appConfig.view.dateformat`

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**